sdk/core/azure-core/LICENSE

```diff
@@ -1,21 +0,0 @@
-    MIT License
```

Comment:
Unlike Embedded C, we do ship "individual packages" in C++ when publishing to vcpkg. I wonder if the better fix is to keep the duplicate LICENSE file for all packages in their individual directories.

For context: https://github.com/Azure/azure-sdk-for-c/pull/1960#discussion_r742431357

@weshaggard and @RickWinter what do you think?

Reply:
For vcpkg, we do take the one from the root of the repo and install in with each package with the name that vcpkg expects: https://github.com/Azure/azure-sdk-for-cpp/blob/main/cmake-modules/AzureVcpkg.cmake#L71-L72

Reply:
I would say we keep only the one at the root given we can copy it into the vcpkg output when we produce the package.

Reply:
This is different than all the other SDKs.  Why do we want to break from standard SDKs guidlines for C++.  C++ has packages

Reply:
The standard is to have one license file in the root of the repo. We only duplicate the file in cases where the source layout is interesting. For example in .NET repo we only have the one license file in the root of the repo.

Reply:
Oh interesting, and good to know.  Would make sense to add EngSys process to copy the file before removing it from everywhere.  I know we don't have process to copy the file into the package.  Granted the package is bigger than it should be right now.

Reply:
We do copy the license from the root of the repo with each vcpackage. See the comments above for details. LICENSE files in each folder are not utilized. Some services/packages don't even have them (Storage).

Reply:
@RickWinter would you approve this PR given the information above?